mlir/lib/Dialect/MIOpen/utility/transformMapUtils.cpp

```diff
@@ -241,15 +241,23 @@ namespace {
 /// next part of memory, etc.), which is an optimistic guess, and the
 /// coefficient that the dimension needs to be multiplied by for the dimension
 /// to vectorize properly when it is embeded into a broader context.
+/// The alignment field tracks what we know about how the values in a particular
+/// dimension will be aligned, with the initial assiumption being that the
+/// input dimension is aligned to the maximum vectorization length. This
+/// information is important in ensuring that a pad() that would otherwise
+/// vectorize doesn't have reads that go partly into the padding.
 struct VectorizationInfo {
-  int64_t maxLength = 0, needsCoefficient = 0;
+  int64_t maxLength = 0, needsCoefficient = 0, alignment = 0;
```

Comment:
I think the base case, if a user doesn't provide length and alignment should that be both 1 for length and alignment.

Reply:
These are struct defaults, no one will see them, if they show up we have a bug